src/Fractions/Fraction.CompareTo.cs

```diff
@@ -41,48 +43,142 @@ namespace Fractions;
+            }
+
+            if (numerator2.IsZero) {
+                return 1; // other is NaN
+            }
 
-        if (numerator2.IsZero) {
-            return IsPositive ? 1 : -1;
+            // comparing infinities
+            return numerator1.Sign.CompareTo(numerator2.Sign);
         }
 
         if (denominator1.IsZero) {
-            return numerator1.Sign; // PositiveInfinity -> 1, NegativeInfinity -> -1
+            return numerator1.Sign switch {
+                1 => 1, // PositiveInfinity -> 1
+                _ => -1 // NaN or NegativeInfinity -> -1
+            };
         }
 
         if (denominator2.IsZero) {
-            return -numerator2.Sign; // PositiveInfinity -> -1, NegativeInfinity -> 1
+            return numerator2.Sign switch {
+                1 => -1, // PositiveInfinity -> -1
+                _ => 1 // NaN or NegativeInfinity -> 1
+            };
         }
 
-        // both values are non-zero fractions with different denominators  
-        return MultiplyTerms(ref numerator1, ref denominator2)
-            .CompareTo(MultiplyTerms(ref numerator2, ref denominator1));
+        var firstNumeratorSign = numerator1.Sign;
+        var secondNumeratorSign = numerator2.Sign;
+
+        if (firstNumeratorSign != secondNumeratorSign) {
+            return firstNumeratorSign.CompareTo(secondNumeratorSign);
+        }
+
+        // both values are non-zero fractions with different denominators
+        if (denominator1 < denominator2) {
+            // reverse the comparison from x.CompareTo(y) to y.CompareTo(x)
+            return firstNumeratorSign == 1
+                ? -comparePositiveTerms(numerator2, denominator2, numerator1, denominator1)
+                : -compareNegativeTerms(numerator2, denominator2, numerator1, denominator1);
+        }
+
+        return firstNumeratorSign == 1
+            ? comparePositiveTerms(numerator1, denominator1, numerator2, denominator2)
+            : compareNegativeTerms(numerator1, denominator1, numerator2, denominator2);
+
+        static int comparePositiveTerms(BigInteger numerator1, BigInteger denominator1, BigInteger numerator2, BigInteger denominator2) {
+            // [denominator1 > denominator2 > 0]
+            // example: {10/10} and {1/1} or {10/100} and {1/10}
+            if (numerator1 <= numerator2) {
+                return -1; // expecting: 0 < numerator2 < numerator1
+            }
+
+            if (numerator2.IsOne) {
+                return denominator2.IsOne ? numerator1.CompareTo(denominator1) : (numerator1 * denominator2).CompareTo(denominator1);
+            }
+
+            if (denominator2.IsOne) {
+                return numerator1.CompareTo(numerator2 * denominator1);
+            }
+
+            // comparing the positive term ratios:
+            // {9/7} / {4/3} = {(2 + 1/4) / (2 + 1/3)} = {(9/4)/(7/3)} = {27/28} => (2).CompareTo(2) == 0 and (1/4).CompareTo(1/3) == -1

```

Comment:
Can you please explain how this comment should be understood?

```
{9/7} / {4/3} = (1 + 2/7) / (1 + 1/3) => (1).CompareTo(1) == 0 AND (2/7).CompareTo(1/3) == -1
```

Reply:
I'm doing the same roundabout comparison again- dividing the two fractions (like with the `Equals`) and then checking if their ratio is larger or smaller than {1/1}.
Another way of thinking about these divisions is to _imagine_ that the terms represent decimals: {9.0 / 7.0} / {4.0 / 3.0} = {(9.0/4.0) / (7.0/3.0) = {2.25 / 2.33} . The part preceding the decimal separator (`2`) is the quotient, while the rest is the `remainder divided by the dividend` (e.g. 1/4 adjusted to denominator of 100 is 25/100).